app/jobs/update_ticket_links_job.rb

```diff
@@ -59,9 +80,12 @@ def post_error_status(full_repo_name, sha)
     CommitStatus.new(full_repo_name: full_repo_name, sha: sha).error
   end
 
-  def commit_on_master?(full_repo_name, sha)
-    git_repo = GitRepositoryLoader.from_rails_config.load(full_repo_name.split('/')[1], update_repo: true)
+  def extract_ticket_key_from_branch_name(branch_name)
+    /(?<ticket_key>[A-Z]{2,8}-[1-9][0-9]*)/ =~ branch_name
+    ticket_key
+  end
 
-    git_repo.commit_on_master?(sha)
+  def url_for_repo_and_sha(full_repo_name, sha)
+    CommitStatus.new(full_repo_name: full_repo_name, sha: sha).target_url
```

Comment:
@sgerrand I took a look at improving the URL creation, but struggled a bit as we do it in so many different ways across this codebase.

This is the solution I came to in the end, but it's still a WIP as this only provides the path and not the host, therefore we'd need a way of feeding the host into the job. Tbf, explicitly handing the host to the job probably makes sense, as it's probably not safe to use the job executors host to create url?

```suggestion
    repo_name = full_repo_name.split('/').last
    Factories::FeatureReviewFactory.new.create_from_apps({ repo_name => sha }).path
```

Hope that makes sense, catch me before 5:30pm if it doesn't. Otherwise, do whatever you feel fit whilst I'm away.